custom_io.c

```diff
@@ -0,0 +1,139 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include "memtable.h"
+#include "custom_io.h"
+#include "error.h"
+
+#define MARKER -1
+#define MAX_LINE_SIZE 100
+#define BUF_DATA_SIZE 50
+
+/* Length is based on structure of filename below... time returns 10 digit number,
+ * plus underscore and another 2 digits  + file suffix (3 char)  + \0 = 17 chars */
+#define FILENAME_SIZE 20
+
+/* prototypes for static functions */
+static void serialize_preorder(MNode *root, FILE *fp);
+static void add_to_file(char *filename, char *to_write);
+static void inorder_to_file(MNode *root, FILE *fp);
+
+/* Writes a memtable (binary tree) to file */
+void serialize_memtable(Memtable *memtable, char *filename) {
+	FILE *fp;
```

Comment:
np; could be one line ```FILE *fp =  fopen(filename, "w");``` Also, can ```fopen``` return an error?

Reply:
fixed!